grafana-plugin/src/pages/users/Users.tsx

```diff
@@ -162,8 +157,6 @@ class Users extends React.Component<UsersProps, UsersState> {
       });
 
     const { count, results } = userStore.getSearchResult();
-    const showMobileAppScreen: boolean =
-      id !== undefined && id !== 'me' && id === userStore.currentUserPk && store.hasFeature(AppFeature.MobileApp);
```

Comment:
this `id !== 'me'` check would hide the modal for me. When I would go to the user's page, then click "View my profile", this would append an `id=me` query parameter, thereby making `showMobileAppScreen` false.

In addition, I decided to move the `showMobileAppScreen` logic to the `UserSettings` component since we are already passing `userPkToEdit` down to this component

Reply:
Yeah, that's a good point. Hmm I didn't notice the issue of `id !== 'me'`. Also I think it's best that you moved it back there, I didn't spot the `isCurrent` which helps in this case. :+1: